src/huggingface_hub/snapshot_download.py

```diff
@@ -18,6 +18,7 @@ def snapshot_download(
     library_name: Optional[str] = None,
     library_version: Optional[str] = None,
     user_agent: Union[Dict, str, None] = None,
+    token: Union[str, None] = None,
```

Comment:
I felt it was simpler to pass an explicit token instead of `use_auth_token` and then retrieving it, but happy to adjust if the latter is preferred

Reply:
In my opinion it's better to retrieve the token so users don't have to get it themselves (`HfFolder.get_token()` should help you have some simple logic).

Reply:
Also that way we're consistent with `cached_download`, `Repository`,  etc.

Reply:
I would also favor a `use_auth_token` based on how it's handled in other methods!

Reply:
Thanks for the feedback @osanseviero @LysandreJik 🙌 !

I've refactored the PR to use `use_auth_token` instead :)